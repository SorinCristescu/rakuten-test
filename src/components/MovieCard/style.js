import styled from 'styled-components';
import { tablet } from '../../globalStyle/mediaQueries';

export const Card = styled.div`
  width: 140px;
  height: 250px;
  ${tablet} {
    width: 80%;
  }
`;

export const Overlay = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  background: rgb(0, 0, 0);
  background: rgba(0, 0, 0, 0.5);
  width: 100%;
  height: 100%;
  transition: 0.5s ease;
  opacity: 0;

  &:hover {
    opacity: 1;
  }
  > p {
    text-align: center;
    color: yellow;
    font-size: 20px;
    padding: 40px;
  }
  ${tablet} {
    width: 80%;
  }
`;

export const CardBody = styled.div`
  width: 140px;
  height: 210px;
  position: relative;
  &:hover {
    border: 3px solid #fdeb48;
  }
  > img {
    display: block;
    width: 100%;
    height: 100%;
  }
  > img:hover {
    width: 140px;
    height: 210px;
  }
`;

export const CardFooter = styled.div`
  height: 30px;
  padding: 0 5px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  > p {
    width: 30px;
    height: 15px;
    font-size: 16px;
    color: #ffffff;
  }
`;

export const Votes = styled.p`
  color: #fdeb48 !important;
`;

export const Score = styled.p``;

export const MovieTitle = styled.h2`
  color: #ffffff;
  font-size: 18px;
  font-weight: 300;
  margin-top: 15px;
  margin-bottom: 15px;
`;
