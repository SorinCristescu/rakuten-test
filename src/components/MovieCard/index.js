import React from 'react';
import PropTypes from 'prop-types';

// Routing
import { NavLink, withRouter } from 'react-router-dom';

// Styles
import { Card, Overlay, CardFooter, CardBody, Score, Votes } from './style';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUser } from '@fortawesome/free-solid-svg-icons';
import { faStar } from '@fortawesome/free-solid-svg-icons';

const MovieCard = ({ movie }) => {
  const { images, highlighted_score } = movie;
  const { formatted_amount_of_votes, score } = highlighted_score;

  return (
    <Card>
      <NavLink to={'/movie/' + movie.id}>
        <CardBody>
          <img src={images.artwork} />
          <Overlay>
            <p>
              Show
              <br />
              details
            </p>
          </Overlay>
        </CardBody>
      </NavLink>
      <CardFooter>
        <span style={{ fontSize: '14px', color: '#fdeb48' }}>
          <FontAwesomeIcon icon={faStar} />
        </span>
        <Votes>{formatted_amount_of_votes}</Votes>
        <span style={{ fontSize: '14px', color: '#ffffff' }}>
          <FontAwesomeIcon icon={faUser} />
        </span>
        <Score>{score}</Score>
      </CardFooter>
    </Card>
  );
};

// Runtime props type checking
MovieCard.propTypes = {
  movie: PropTypes.object.isRequired,
};

export default withRouter(MovieCard);
