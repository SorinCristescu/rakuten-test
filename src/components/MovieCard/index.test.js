import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './index';

describe('MovieCard', () => {
  it('should be defined', () => {
    expect(MovieCard).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<MovieCard />);
    expect(tree).toMatchSnapshot();
  });
});
