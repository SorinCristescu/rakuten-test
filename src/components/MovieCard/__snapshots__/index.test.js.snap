// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should render correctly 1`] = `ShallowWrapper {}`;
